### PR TITLE
Adding tsid

### PR DESCRIPTION
**Pull request by EtienneAr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jcarpent** [comment]
I do agree to be a maintainer for this feedstock.

**Comment by EtienneAr** [comment]
@conda-forge/help-c-cpp This PR is ready for review.

File: recipes/tsid/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - {{ pin_subpackage(name, max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}  # [not win]
+    - cmake
+    - make  # [not win]
+  host:
+    - eigenpy
+    - boost
```
Comment (chrisburr) [inline recipes/tsid/meta.yaml:25]: I guess you're not using boost python so this should be `boost-cpp`

File: recipes/tsid/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}  # [not win]
+    - cmake
+    - make  # [not win]
+  host:
+    - eigenpy
```
Comment (chrisburr) [inline recipes/tsid/meta.yaml:24]: You need python in `host` and `run` so it builds once per python version as they have incompatible ABIs

**Comment by EtienneAr** [comment]
Thank you @chrisburr for your feedback. I changed it in 6c2d959.

**Pull request merged**